wdns/rdata_to_ubuf.c

```diff
@@ -245,11 +204,75 @@ svcparam_to_str(uint16_t key, const uint8_t *src, uint16_t len, ubuf *u)
 		}
 		break;
 
-	case spr_invalid:
+	/*
+	 * The wire format value for "alpn" consists of at least one * "alpn-id"
+	 * prefixed by its length as a single octet, and these length-value
+	 * pairs are concatenated to form the SvcParamValue. These MUST exactly
+	 * fill the SvcParamValue; otherwise, the SvcParamValue is malformed.
+	 *
+	 * Note that we always wrap such values around double quotes in
+	 * presentation format.
+	 */
+	case spr_alpn:
+		ubuf_add_fmt(u, "\"");
```

Comment:
I think if you are not using printf-style formatting, these ubuf_add_fmt() calls should be ubuf_add_cstr() calls. Which is only mildly important if we are doing a _lot_ of conversions, so maybe not worth changing the code.